### PR TITLE
CBE: promote an already passing test, and add implementation for clz, ctz for integers

### PR DESCRIPTION
**Pull request by jmc-88**

### CODE REVIEW

File: src/codegen/c.zig [hunk 1]
```diff
@@ -3348,6 +3348,31 @@ fn airBuiltinCall(f: *Function, inst: Air.Inst.Index, fn_name: [*:0]const u8) !C
...
+    const local = try f.allocLocal(inst_ty, .Const);
+    const operand = f.air.instructions.items(.data)[inst].ty_op.operand;
+    const operand_ty = f.air.typeOf(operand);
+    const target = f.object.dg.module.getTarget();
+    const writer = f.object.writer();
+
+    const fn_name = switch (which_zeroes) {
+        .leading => "zig_clz",
+        .trailing => "zig_ctz",
+    };
```
Comment (andrewrk) [inline src/codegen/c.zig:3364]: You know what's even better than branching twice? Only branching once! Just pass `fn_name` into `airCountZeroes`:

```zig
     try writer.print(" = zig_{s}(", .{fn_name}); 
```

File: src/link/C/zig.h [hunk 2]
```diff
@@ -500,3 +500,21 @@ zig_shl_sat_s(isize, intptr_t, ((sizeof(intptr_t)) * CHAR_BIT - 1))
 zig_shl_sat_s(short,    short, ((sizeof(short   )) * CHAR_BIT - 1))
 zig_shl_sat_s(int,        int, ((sizeof(int     )) * CHAR_BIT - 1))
 zig_shl_sat_s(long,      long, ((sizeof(long    )) * CHAR_BIT - 1))
+
+#define bitsizeof(T) (CHAR_BIT * sizeof(T))
```
Comment (andrewrk) [inline src/link/C/zig.h:504]: don't pollute the global namespace:

```suggestion
#define zig_bitsizeof(T) (CHAR_BIT * sizeof(T))
```

**Review by andrewrk — CHANGES_REQUESTED** [review]
Nice work!

File: src/codegen/c.zig (hunk 1 above)
Comment (jmc-88) [reply src/codegen/c.zig]: Done.

File: src/link/C/zig.h (hunk 2 above)
Comment (jmc-88) [reply src/link/C/zig.h]: Done.

File: test/behavior/math.zig [hunk 3]
```diff
@@ -76,8 +75,11 @@ fn testClz() !void {
     try expect(testOneClz(u8, 0b00001010) == 4);
     try expect(testOneClz(u8, 0b00011010) == 3);
     try expect(testOneClz(u8, 0b00000000) == 8);
-    try expect(testOneClz(u128, 0xffffffffffffffff) == 64);
-    try expect(testOneClz(u128, 0x10000000000000000) == 63);
+
+    if (builtin.zig_backend != .stage2_c) { // TODO
```
[the remark this answers is not in the document]
Comment (jmc-88) [reply test/behavior/math.zig]: Done.

**Review by andrewrk — APPROVED** [review]

**Pull request merged**